core/divelist.c

```diff
@@ -1287,7 +1287,7 @@ static struct dive *get_last_valid_dive()
  */
 int get_dive_nr_at_idx(int idx)
 {
-	struct dive *last_dive = get_last_valid_dive(dive_table.nr - 1);
+	struct dive *last_dive = get_last_valid_dive();
```

Comment:
The whole function appears to be buggy.
There seems to be a
```
        if (idx < dive_table.nr)
                return 0;
```
missing, isn't it? At least that's what I read from the comment.

Reply:
but that condition is true for all dives. What am I missing?

Reply:
I don't think so - if you add at the end of the table then `idx == dive_table.nr`. However, the error condition is probably a not very common case - you would have to have dives in the future and then add a new dive with the current date, or something. All a bit synthetic.

Reply:
very common case - all manually added dives tend to be in the future by default.
But as I said in my review comment, I need to read the calling code to actually understand what this does

Reply:
Not sure if you're being facetious.

The common case is that the newly added dive (which gets the current timestamp) is the last in the list. If this it isn't the case (which is probably very uncommon), then this code fails.

What you have to do to hit this bug:
1) Add dive (gets current timestamp).
2) Set dive date to the future.
3) Add another dive -> boom wrong number.

Reply:
@bstoeger wrote that in 0d78fe45c3

Reply:
Thanks - I already noticed that. :-D